### PR TITLE
Make lesson functions

### PR DESCRIPTION
**Pull request by Tomoki108**
***Outline***
-lessonAnswer, lessonResult and category page are the views. Those pages are to take lessons.

***Commands to run***
-none

***Preconditions***
-Login with any account.
-Go to (hostname/category) to go to category page. And push one of Start button to take lesson.

***Expected outputs***
-category, lessonAnswer and lessonResult page should be displayed properly with proper contents.
-Proper data should be stored into Lesson and LessonWord model when user finish a lesson. 

***asana***
-https://app.asana.com/0/984197702640917/984490302593960

### CODE REVIEW

**Comment by jeremiahC** [comment]
@Tomoki108 
I have error when i answer lesson
https://gyazo.com/52a588b8b25296e53a48e2141e5068b7

**Comment by Tomoki108** [comment]
@jeremiahC 
Sorry I forgot to write one precondition.
User have to  be logged in when they take lesson. After I make authentication, I'll prohibit to take lesson without login for user. But for now, please test one more time being logged in.

**Comment by jeremiahC** [comment]
@Tomoki108 
please make font size a little bit bigger because it is to small to read

**Comment by Tomoki108** [comment]
@jeremiahC
I got it.

**Comment by jeremiahC** [comment]
@Tomoki108 
in this PR, is there a function that user can't retake lesson or it is in another PR?

**Comment by Tomoki108** [comment]
@jeremiahC 
Sorry I forgot the requirement. I'll make that in this PR.

File: resources/views/lessonAnswer.blade.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <div class="row">
+        <div class="col-md-6 col-sm-12">
+            <h1 class="mt-4">{{ $words->get($page_number)->content }}</h1>
+        </div>
+        <div class="col-md-6 col-sm-12">
+            @foreach($word_answers as $word_answer)
+                @if($word_answer->correct == 1)
+                    <a href="{{ route($route, ['category' => $category->id, 'page_number' => $page_number+1, 'correct' => 1]) }}">
+                @else
+                    <a href="{{ route($route, ['category' => $category->id, 'page_number' => $page_number+1, 'correct' => 0]) }}">
```
Comment (jeremiahC) [inline resources/views/lessonAnswer.blade.php]: i think this condition is too long for just to change the value of correct index. you can use the shorthand if else. like this:

```
<a href="{{ route($route, ['category' => $category->id, 'page_number' => $page_number+1, 'correct' => $word_answers->correct == 1 ? 1 :0]) }}">
    <button class="btn btn-primary mt-4 w-100">{{ $word_answer->content }}</button>
</a>
```

File: resources/views/lessonResult.blade.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+    //Processing user answers data to display.
+    $result = session()->get('result');
+    $success = (in_array('1', $result)) ? array_count_values($result)[1] : 0;
+    extract($result, EXTR_PREFIX_INVALID, 'answer');
+?>
```
Comment (jeremiahC) [inline resources/views/lessonResult.blade.php:4]: it's better that controller will handle this code so blade will be cleaner.

Comment (Tomoki108) [reply resources/views/lessonResult.blade.php:4]: @jeremiahC 
I could not help using extract function in the view, but others are removed.

File: resources/views/lessonAnswer.blade.php (hunk 1 above)
Comment (Tomoki108) [reply resources/views/lessonAnswer.blade.php]: Thanks, I fixed it.

**Review by justinPCM — COMMENTED** [review]
- When adding words to a category, it's better to limit the text input for bot the word itself and the choices. Overly huge text inputs will destroy your UI.
- When I click a category without any words yet it will say "This lesson has no contents yet", but when I add a word on that same category and go back to the category page again, I can take the test already but it still says "This lesson has no contents yet"

**Review by justinPCM — APPROVED** [review]

**Pull request merged**